[Shem] iP

sheimoria opened the pull request:
# Duke (Shem's Version)

> “It is only with the heart that one can see rightly; what is essential is invisible to the eye.” – Antoine de Saint-Exupéry,  [The Little Prince](https://en.wikipedia.org/wiki/The_Little_Prince)

*Daddy* Duke frees your mind of having to remember all the things you need to do. It's

- Text-based
- Easy to learn
- ~~Fast~~ Super fast to use
- Completely **free**!

## Oh my days how do I get it?

1. Download it from [here](https://github.com/nus-cs2103-AY2122S2/ip/pull/193).
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you 😉

## What can this baby do?

- [ ] Manage to-dos, deadlines and events
- [ ] Find tasks with a specific keyword

If you're a Java programmer, you can use it to practice Java too. Here's a snippet of the `main` method:

```java
public void run() {
     ui.showWelcome();
     boolean isExit = false;
     while (!isExit) {
         try {
             String fullCommand = ui.readCommand();
             Command c = Parser.parse(fullCommand);
             c.execute(storage, tasks, ui);
             isExit = c.isExit();
         } catch (DukeException e) {
             ui.showError(e.getMessage());
         }
     }
 }

 public static void main(String[] args) {
     new Duke("data/tasks.txt").run();
 }
```

Commit:
Add Gradle support

sheimoria committed:
feat: greet, echo, exit

sheimoria committed:
feat: add, list

sheimoria committed:
feat: mark as done

sheimoria committed:
feat: todos, events, deadlines

sheimoria committed:
feat: automated text ui testing

sheimoria committed:
feat: handle errors

sheimoria committed:
feat: delete

sheimoria committed:
feat: save

sheimoria committed:
feat: date and time

sheimoria committed:
Merge branch 'branch-Level-8'
feat: add date and time to tasks

sheimoria committed:
fix: date and time

sheimoria committed:
feat: use more OOP

sheimoria committed:
feat: organise into packages

sheimoria committed:
feat: add junit tests

sheimoria committed:
feat: add javadoc comments

sheimoria committed:
fix: coding standard

sheimoria committed:
feat: find

sheimoria committed:
fix: merge conflict

sheimoria committed:
fix: merge conflict

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,17 @@
 public class Duke {
     private static ArrayList<Task> tasks = new ArrayList<>();
 
+    private static String isPlural(int n) {
```
I think it may be better if this method is renamed, since it may be confused with one that returns a boolean true or false value.

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -28,11 +28,11 @@ private static void load() throws IOException, DukeException {
             switch(type) {
```
Should have a space before the bracket i.e `switch (type)`. Just a minor code style thing though

geraldneo567 commented on src/main/java/Duke.java:
```diff
@@ -28,11 +28,11 @@ private static void load() throws IOException, DukeException {
-                    tasks.add(new Event(description, status, at));
+                    tasks.add(new Event(description, at, status));
                     break;
                 case "D":
```
I believe the coding standard states no indentation for case clauses in switch cases. Should not affect the code though!

geraldneo567 commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,16 @@
+
+import duke.task.ToDo;
+import org.junit.jupiter.api.Test;
+
```
I believe the static imports should be at the top based on coding standard (although not 100% sure)

sheimoria committed:
refactor: Storage load method

sheimoria committed:
Merge remote-tracking branch 'origin/add-gradle-support'

sheimoria committed:
fix: task list unable to print when there are no tasks

sheimoria committed:
feat: set up gradle

sheimoria committed:
feat: use checkstyle

sheimoria committed:
feat: add GUI

sheimoria committed:
feat: update styles

sheimoria committed:
refactor: file structure

sheimoria committed:
feat: use assertions

sheimoria committed:
refactor: improve code quality

sheimoria committed:
feat: set up CI

sheimoria committed:
refactor: improve code quality

sheimoria committed:
Revert "feat: set up CI"

This reverts commit 6f94c1e911f7f121253c668f311c265e803f5246.

sheimoria committed:
Revert "Revert "feat: set up CI""

This reverts commit 9bb0203b0d35f96460b8fb8aa92e9abec8425d1d.

sheimoria committed:
Revert "refactor: improve code quality"

This reverts commit 8bd10cca292e42d8fcf6fc65707aeb1aa7dad937.

sheimoria committed:
Revert "feat: use assertions"

This reverts commit 7eae4eb32ee72d8d9b3d706c01dac9327929b960.

sheimoria committed:
feat: use assertions

sheimoria committed:
Delete .github directory

sheimoria committed:
Merge pull request #1 from sheimoria/branch-A-Assertions

feat: use assertions

sheimoria committed:
fix: merge conflict with master

sheimoria committed:
Merge pull request #2 from sheimoria/branch-A-CodeQuality

refactor: improve code quality

sheimoria committed:
feat: set up CI

sheimoria committed:
feat: set up CI

There is no continuous integration configured for the software.

Continuous integration, which is the practice of automating
integration of code changes from multiple contributors into a
single software project.

A primary DevOps best practice, it would allow us to run builds
and tests on frequent merges of code changes. Automated tools
would help assert the new code's correctness before integration.

Let's implement a basic workflow to check whether pull requests
are functional builds.

Implementing such a system would prevent us from merging
incorrect code that would break our software.

Refer to this article by Atlassian to learn more about continuous
integration:
https://www.atlassian.com/continuous-delivery/continuous-integration

sheimoria committed:
Merge branch 'master' of https://github.com/sheimoria/duke

sheimoria committed:
feat: add priority

There is no way to set priority levels for tasks.

Being able to set a priority level for each task will allow users to
note down which tasks are more urgent and therefore should be done first.

Let's add the ability to set a priority level for each task by
integrating into the Task class an Priority enum with three values:
* LOW
* MEDIUM
* HIGH

Using an enum is suitable because priority levels can be defined as a
set of predefined values, and using it over, say, a range of integer
values will also make the code easier to understand.

Refer to this website for more information on enum:
https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

sheimoria committed:
Merge pull request #3 from sheimoria/branch-C-Priority

feat: add priority

sheimoria committed:
feat: improve GUI

sheimoria committed:
Merge pull request #5 from sheimoria/branch-A-BetterGUI

feat: improve GUI

sheimoria committed:
feat: add user guide

sheimoria committed:
Merge pull request #6 from sheimoria/branch-A-UserGuide

feat: add user guide

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Set theme jekyll-theme-midnight

sheimoria committed:
Merge branch 'master' of https://github.com/sheimoria/ip

sheimoria committed:
feat: add user guide

sheimoria committed:
Merge pull request #7 from sheimoria/branch-A-UserGuide

feat: add user guide

sheimoria committed:
feat: add help dialog

sheimoria committed:
feat: add more Parser classes
